cli/utils.py

```diff
@@ -249,33 +249,98 @@ def print_job(client, job: aztk.spark.models.Job):
     print_format = '{:<36}| {:<15}'
 
     log.info("")
-    log.info("Job              %s", job.id)
+    log.info("Job               %s", job.id)
     log.info("------------------------------------------")
-    log.info("State:           %s", job.state)
-    log.info("Transition Time: %s", utc_to_local(job.state_transition_time))
+    log.info("State:            %s", job.state)
+    log.info("Transition Time:  %s", utc_to_local(job.state_transition_time))
     log.info("")
 
     if job.cluster:
-        print_cluster(client, job.cluster)
+        print_cluster_summary(job.cluster)
+    else:
+        if job.state == 'completed':
+            log.info("Cluster           %s", "Job completed, cluster deallocated.")
+            log.info("")
+        else:
+            log.info(print_format.format("Cluster", "Provisioning"))
+            log.info("")
 
-    print_applications(job.applications)
+    if job.applications:
+        application_summary(job.applications)
+    else:
+        application_summary(client.list_applications(job.id))
     log.info("")
 
 
-def print_applications(applications: List[aztk.spark.models.Application]):
+def node_state_count(cluster: aztk.spark.models.Cluster):
+    states = {}
+    for state in batch_models.ComputeNodeState:
+        states[state] = 0
+    for node in cluster.nodes:
+        states[node.state] += 1
+    return states
+
+
+def print_cluster_summary(cluster: aztk.spark.models.Cluster):
+    print_format = '{:<4} {:<23} {:<15}'
+
+    log.info("Cluster           %s", cluster.id)
+    log.info("-"*42)
+    log.info("Nodes             %s", __pretty_node_count(cluster))
+    log.info("| Dedicated:      %s", __pretty_dedicated_node_count(cluster))
+    log.info("| Low priority:   %s", __pretty_low_pri_node_count(cluster))
+    state_count = node_state_count(cluster)
```

Comment:
One last change I'd add here: If the master is not yet available, I would show that, and then in the application_summary print method I would make a statement that no applications are scheduled until a master node is assigned.

Reply:
Fixed. Now, if the master is not selected a warning is logged below the job and application summaries.